streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCache.java

```diff
@@ -105,14 +100,28 @@ public boolean persistent() {
     }
 
     @Override
-    public V get(K key) {
-        return this.map.get(key);
+    public boolean open() {
+        return open;
+    }
+
+    @Override
+    public  V get(K key) {
+        lock.readLock().lock();
+        try {
+            return this.map.get(key);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     @Override
     public void put(K key, V value) {
-        this.map.put(key, value);
-        this.keys.add(key);
+        lock.writeLock().lock();
```

Comment:
CLHM was the baseline algorithm for Guava, though is much faster due to leaving G before optimizing the port. If you later investigate this in-depth, I'd suggest Caffeine now since it includes a superior eviction policy and tons of features. Cheers.